diff --git a/node_modules/@sefaria/search/search.js b/node_modules/@sefaria/search/search.js
index ccc8c96..039dd4f 100644
--- a/node_modules/@sefaria/search/search.js
+++ b/node_modules/@sefaria/search/search.js
@@ -7,8 +7,9 @@ const FilterNode = require('./FilterNode');
 const SearchState = require('./searchState');
 
 class Search {
-    constructor(apiHost, searchIndexText, searchIndexSheet) {
+    constructor(apiHost, searchIndexText, searchIndexSheet, extraHeaders = {}) {
       this.apiHost = apiHost;
+      this.extraHeaders = extraHeaders;
       this.searchIndexText = searchIndexText;
       this.searchIndexSheet = searchIndexSheet;
       this._cache = {}
@@ -47,6 +48,7 @@ class Search {
             method: 'POST',
             body: req,
             headers: {
+              ...this.extraHeaders,
               'Accept': 'application/json',
               'Content-Type': 'application/json; charset=utf-8',
             },
